src/plugin/luapi_application.h

```diff
@@ -475,6 +476,194 @@ static int applib_changeBackgroundPdfPageNr(lua_State* L) {
+    ToolHandler* toolHandler = control->getToolHandler();
+
+    std::string size = toolSizeToString(toolHandler->getPenSize());
+    int fillOpacity = toolHandler->getPenFill();
+    bool fill = toolHandler->getPenFillEnabled();
+
+    lua_newtable(L);
+
+    lua_pushliteral(L, "size");
+    lua_pushstring(L, size.c_str());
+    lua_settable(L, -3);
+
+    lua_pushliteral(L, "fill");
+    lua_pushboolean(L, fill);
+    lua_settable(L, -3);
+
+    lua_pushliteral(L, "fillOpacity");
+    lua_pushinteger(L, fillOpacity);
+    lua_settable(L, -3);
+
+    return 1;
+}
+
+/*
+ * Returns a table encoding all info on the highlighter in a Lua table of the shape
+ * {
+ *   "size" = string
+ *   "fill" = bool
+ *   "fillOpacity" = integer
+ * }
+ *
+ * See /src/control/ToolEnums.cpp for possible values of "size".
+ *
+ * Example 1: local highlighterInfo = app.getHighlighterInfo()
+ *            local size = highlighterInfo["size"]
+ *            local opacity = highlighterInfo["fillOpacity"]
+ *
+ * Example 2: local fill = app.getHighlighterInfo()["fill"]
+ */
+static int applib_getHighlighterInfo(lua_State* L) {
+    Plugin* plugin = Plugin::getPluginFromLua(L);
+    Control* control = plugin->getControl();
+    ToolHandler* toolHandler = control->getToolHandler();
+
+    std::string size = toolSizeToString(toolHandler->getHighlighterSize());
+    int fillOpacity = toolHandler->getHighlighterFill();
+    bool fill = toolHandler->getHighlighterFillEnabled();
+
+    lua_newtable(L);
+
+    lua_pushliteral(L, "size");
+    lua_pushstring(L, size.c_str());
+    lua_settable(L, -3);
+
+    lua_pushliteral(L, "fill");
+    lua_pushboolean(L, fill);
+    lua_settable(L, -3);
+
+    lua_pushliteral(L, "fillOpacity");
+    lua_pushinteger(L, fillOpacity);
+    lua_settable(L, -3);
+
+    return 1;
+}
+
+/*
+ * Returns a table encoding all info on the eraser in a Lua table of the shape
+ * {
+ *   "type" = string
+ *   "size" = string
+ * }
+ *
+ * See /src/control/ToolEnums.cpp for possible values of "type" and "size".
+ *
+ * Example 1: local eraserInfo = app.getEraserInfo()
+ *            local type = eraserInfo["type"]
+ *            local size = eraserInfo["size"]
+ *
+ * Example 2: local type = app.getEraserInfo()["type"]
+ */
+static int applib_getEraserInfo(lua_State* L) {
+    Plugin* plugin = Plugin::getPluginFromLua(L);
+    Control* control = plugin->getControl();
+    ToolHandler* toolHandler = control->getToolHandler();
+
+    std::string type = eraserTypeToString(toolHandler->getEraserType());
+    std::string size = toolSizeToString(toolHandler->getEraserSize());
+
+    lua_newtable(L);
+
+    lua_pushliteral(L, "type");
+    lua_pushstring(L, type.c_str());
+    lua_settable(L, -3);
+
+    lua_pushliteral(L, "size");
+    lua_pushstring(L, size.c_str());
+    lua_settable(L, -3);
+
+    return 1;
+}
+
+
+/*
+ * Returns a table encoding all info on the active tool in a Lua table of the shape
+ * {
+ *   "type" = string
+ *   "size" = string
```

Comment:
It might make more sense to make `thickness` a part of size:
```lua
  size = {
    name = string,
    value = number,
  },
```

Although we're currently using `size` to refer to an enum value in the codebase and `thickness` to refer to the corresponding line thickness, I think grouping them might make the API easier to understand.

Possible issue: If we add `app.setActiveToolInfo(info)`, a size of ` { name = small, thickness = 1e8 }` could be invalid. Does it set the thickness to `1e8` or use the thickness associated with `small`?

Reply:
I need to take a look at the code of what name/value pairs are currently possible and what #3288 will change before I can answer on this.

Reply:
@personalizedrefrigerator I agree with the idea to make `thickness` part of size and applied it in my last commit. Moreover I added the info about the thickness of the pen, highlighter and eraser.
The issues you mentioned don't look too critical to me.
- There is no need to add `app.setActiveToolInfo`, since those properties can (and should) be set through `app.uiAction` and `app.changeToolColor`. The thickness is an exception and we will have to add API to set the custom thickness implemented in #3288 
- Looking at the current implementation of #3288 suggests to me that getting the info on the individual tool sizes will continue to work as expected.